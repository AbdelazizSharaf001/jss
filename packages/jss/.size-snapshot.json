{
  "dist/jss.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  },
  "dist/jss.min.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  }
}
