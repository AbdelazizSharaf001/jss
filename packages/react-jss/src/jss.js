// @flow
import {create} from 'jss'
import preset from 'jss-preset-default'

export const jss = create(preset())
