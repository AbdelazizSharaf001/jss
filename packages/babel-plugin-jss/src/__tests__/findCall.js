import {stripIndent} from 'common-tags'
import {sheets} from 'jss'
import {transform} from '../testUtils'

describe('find function call', () => {
  beforeEach(() => {
    sheets.reset()
  })

  test('support default createStyleSheet identifier', () => {
    const code = stripIndent`
      import jss from 'jss';
      jss.createStyleSheet({
        a: {
          color: 'red'
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('support package name with path to a module', () => {
    const code = stripIndent`
      import jss from 'jss/something';
      jss.createStyleSheet({
        a: {
          color: 'red'
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('accept custom identifier names over configuration', () => {
    const code = stripIndent`
      import xyz from 'x';
      xyz({
        a: {
          color: 'red'
        }
      });
    `
    const options = {identifiers: [{package: /x/, functions: ['xyz']}]}
    expect(transform(code, options)).toMatchSnapshot()
  })

  test('decorators', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';

      @injectSheet({
        a: {
          color: 'red'
        }
      })
      class A {}

      ;
    `
    expect(transform(code)).toMatchSnapshot()
  })
})
