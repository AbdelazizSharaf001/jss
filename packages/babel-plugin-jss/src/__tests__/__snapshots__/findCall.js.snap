// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find function call accept custom identifier names over configuration 1`] = `
"[v6]
import xyz from 'x';
xyz({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import xyz from 'x';
xyz({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`find function call decorators 1`] = `
"[v6]
import injectSheet from 'react-jss';

@injectSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, {
  'classes': {
    'a': 'a-id'
  }
})
class A {}

;
[v7]
import injectSheet from 'react-jss';

@injectSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
})
class A {}

;"
`;

exports[`find function call support default createStyleSheet identifier 1`] = `
"[v6]
import jss from 'jss';
jss.createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
jss.createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`find function call support package name with path to a module 1`] = `
"[v6]
import jss from 'jss/something';
jss.createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss/something';
jss.createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;
